### PR TITLE
feat: add 2D Composite, Intersection and Polygon2d geometry methods

### PR DESCRIPTION
**Pull request by Derollez**
First wrapping of Polygon2d intersection and difference operations based on Boost:
- Use `boost::geometry::intersection` for compute intersection between 2D `Polygon` objects. Add `intersectionWith` method to `Polygon` class in 2D.
- Modify `Polygon::Impl::Impl` constructor to correct `boost::geometry::polygon` instead of rings alone in order to always store `valid` (in the boost sense) `polygon_` objects that can then be leveraged by `intersection` and `difference` methods. 
- Add `Composite` class in 2D (similar to concept on 3D) mainly to store a set of heterogeneous 2D Geometry Objects and be leveraged by `Intersection`. Will be useful later as well

### CODE REVIEW

**Review by vishwaLoft — COMMENTED** [review]
looks good to me!

**Review by vishwaLoft — COMMENTED** [review]
Awesome, looks good!

**Comment by Derollez** [comment]
Screen Shot 2022-10-05 at 3 24 08 PM
Screen Shot 2022-10-05 at 3 21 56 PM
Screen Shot 2022-10-05 at 3 21 16 PM
Screen Shot 2022-10-05 at 3 20 24 PM
Screen Shot 2022-10-05 at 3 19 12 PM
Screen Shot 2022-10-05 at 3 18 52 PM
Screen Shot 2022-10-05 at 2 47 41 PM
Screen Shot 2022-10-05 at 2 47 25 PM
Screen Shot 2022-10-05 at 2 39 06 PM

Tested with latest Astro in Jupyter with a couple of examples

File: bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Intersection.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        .def("as_composite", +[] (const Intersection& anIntersection) -> Composite { return anIntersection.as<Composite>() ; })
+
+        .def("access_composite", &Intersection::accessComposite, return_value_policy::reference)
+
+        .def("get_type", &Intersection::getType)
+
+        // Define static methods
+        .def_static("undefined", &Intersection::Undefined)
+        .def_static("empty", &Intersection::Empty)
+        .def_static("point", &Intersection::Point)
```
Comment (lucas-bremond) [inline bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Intersection.cpp:70]: Any chance you could add the `arg` definitions to the various bindings (this is something we started to do, to be more Python friendly)

File: bindings/python/test/geometry/d2/objects/test_composite.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    composite = Composite(polygon)
+    composite += Composite(point)
+
+    return composite
+
+################################################################################################################################################################
+
+class TestComposite:
+
+    def test_geometry_d2_objects_composite_constructor (
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_composite.py:51]: No need for `geometry_d2_objects_composite` since implied by file path + class name; and add `_success` suffix to indicate this is a success-case test.

```suggestion
    def test_constructor_success (
```

File: bindings/python/test/geometry/d2/objects/test_composite.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    ):
+
+        assert composite.get_object_count() == 2
+        assert composite.access_object_at(0) == polygon
+        assert composite.access_object_at(1) == point
+
+        with pytest.raises(RuntimeError):
+            composite.access_object_at(2)
+
+    # def test_geometry_d2_objects_composite_contains (
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_composite.py:144]: TBR?

File: bindings/python/test/geometry/d2/objects/test_composite.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+Object = mathematics.geometry.d2.Object
+Point = mathematics.geometry.d2.objects.Point
+Polygon = mathematics.geometry.d2.objects.Polygon
+Composite = mathematics.geometry.d2.objects.Composite
+Segment = mathematics.geometry.d2.objects.Segment
+
+################################################################################################################################################################
+
+@pytest.fixture
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_composite.py:24]: These fixtures could be added to a `conftest.py` at `d2` level to avoid multiplying the defs across files?

File: include/OpenSpaceToolkit/Mathematics/Geometry/3D/Objects/Polygon.hpp [hunk 5]
```diff
@@ -46,6 +46,7 @@ using ostk::math::geom::d3::objects::Point ;
 using ostk::math::geom::d3::objects::Segment ;
 using ostk::math::geom::d3::objects::LineString ;
 using ostk::math::geom::d3::objects::Plane ;
+using ostk::math::geom::d3::Intersection ;
```
Comment (lucas-bremond) [inline include/OpenSpaceToolkit/Mathematics/Geometry/3D/Objects/Polygon.hpp:49]: TBR since not used?

File: bindings/python/test/geometry/d2/objects/test_polygon.py [hunk 6]
```diff
@@ -352,4 +353,75 @@ def square (center: Point) -> Polygon:
...
+            '''
+            Creates and Return a Square centered in center
+            and with length 1.0 with no inner rings.
+            '''
+            point_1: Point = center + [0.5, 0.5]
+            point_2: Point = center + [0.5, -0.5]
+            point_3: Point = center + [-0.5, -0.5]
+            point_4: Point = center + [-0.5, 0.5]
+
+            return Polygon([point_1, point_2, point_3, point_4])
```
[the remark this answers is not in the document]
Comment (robinpdm) [reply bindings/python/test/geometry/d2/objects/test_polygon.py:405]: Move it outside test / make it a fixture, to avoid duplication?

File: src/OpenSpaceToolkit/Mathematics/Geometry/2D/Objects/LineString.cpp [hunk 7]
```diff
@@ -61,6 +61,48 @@ bool                            LineString::isEmpty                         ( )
...
+    if (pointCount == 1)
+    {
+        return (this->accessPointAt(0) == aPoint) ;
+    }
+
+    else
+    {
+        for (int index = 0; index < (pointCount- 1); index++)
+        {
+            Point firstPoint = this->accessPointAt(index) ;
```
Comment (robinpdm) [inline src/OpenSpaceToolkit/Mathematics/Geometry/2D/Objects/LineString.cpp:88]:
```suggestion
    }
    else
    {
        for (int index = 0; index < (pointCount- 1); index++)
        {

            Point firstPoint = this->accessPointAt(index) ;
```

File: test/OpenSpaceToolkit/Mathematics/Geometry/2D/Objects/Polygon.test.cpp [hunk 8]
```diff
@@ -1220,18 +1231,432 @@ TEST (OpenSpaceToolkit_Mathematics_Geometry_2D_Objects_Polygon, GetConvexHull)
...
+
+        Intersection intersection = polygon_1.intersectionWith(polygon_2) ;
+
+        EXPECT_TRUE(intersection.isDefined()) ;
+        EXPECT_TRUE(intersection.isEmpty()) ;
+        EXPECT_TRUE(intersection.accessComposite().getObjectCount() == 0) ;
+
+    }
+
+    // Line String intersection between two intersecting convex polygons
```
Comment (robinpdm) [inline test/OpenSpaceToolkit/Mathematics/Geometry/2D/Objects/Polygon.test.cpp:1464]:
```suggestion
    // Line string intersection between two intersecting convex polygons
```

File: bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Objects.cpp [hunk 9]
```diff
@@ -8,6 +8,7 @@
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 #include <OpenSpaceToolkitMathematicsPy/Geometry/2D/Objects/MultiPolygon.cpp>
+#include <OpenSpaceToolkitMathematicsPy/Geometry/2D/Objects/Composite.cpp>
```
Comment (robinpdm) [inline bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Objects.cpp:11]: To keep the ordering consistent w/ below:
```suggestion
#include <OpenSpaceToolkitMathematicsPy/Geometry/2D/Objects/Composite.cpp>
#include <OpenSpaceToolkitMathematicsPy/Geometry/2D/Objects/MultiPolygon.cpp>
```

File: bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Objects/MultiPolygon.cpp [hunk 10]
```diff
@@ -28,7 +28,7 @@ inline void                     OpenSpaceToolkitMathematicsPy_Geometry_2D_Object
 
     class_<MultiPolygon, Object>(aModule, "MultiPolygon")
 
-        .def(init<const Array<Polygon>&>())
+        .def(init<const Array<Polygon>&>(), arg("polygon_array"))
```
Comment (lucas-bremond) [inline bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Objects/MultiPolygon.cpp:31]:
```suggestion
        .def(init<const Array<Polygon>&>(), arg("polygons"))
```

File: bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Objects/LineString.cpp [hunk 11]
```diff
@@ -25,7 +25,7 @@ inline void                     OpenSpaceToolkitMathematicsPy_Geometry_2D_Object
 
     class_<LineString, Object>(aModule, "LineString")
 
-        .def(init<const Array<Point>&>())
+        .def(init<const Array<Point>&>(), arg("point_array"))
```
Comment (lucas-bremond) [inline bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Objects/LineString.cpp:28]:
```suggestion
        .def(init<const Array<Point>&>(), arg("points"))
```

File: bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Objects/PointSet.cpp [hunk 12]
```diff
@@ -25,7 +25,7 @@ inline void                     OpenSpaceToolkitMathematicsPy_Geometry_2D_Object
 
     class_<PointSet, Object>(aModule, "PointSet")
 
-        .def(init<const Array<Point>&>())
+        .def(init<const Array<Point>&>(), arg("point_array"))
```
Comment (lucas-bremond) [inline bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Objects/PointSet.cpp:28]:
```suggestion
        .def(init<const Array<Point>&>(), arg("points"))
```

File: bindings/python/test/geometry/d2/objects/test_composite.py [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+Object = mathematics.geometry.d2.Object
+Point = mathematics.geometry.d2.objects.Point
+Polygon = mathematics.geometry.d2.objects.Polygon
+Composite = mathematics.geometry.d2.objects.Composite
+
+################################################################################################################################################################
+
+class TestComposite:
+
+    def test_geometry_d2_objects_composite_constructor (
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_composite.py:25]:
```suggestion
    def test_constructor_success (
```

File: bindings/python/test/geometry/d2/objects/test_line.py [hunk 14]
```diff
@@ -69,12 +61,10 @@ def test_undefined_success (self):
...
 
-    def test_points_success (self):
-
-        line = Line.points(
-            Point(1.0, 2.0),
-            Point(2.0, 4.0),
-        )
+    def test_points_success (
+        self,
+        line: Line
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_line.py:66]:
```suggestion
        line: Line,
```

File: bindings/python/test/geometry/d2/objects/test_line.py [hunk 15]
```diff
@@ -36,26 +36,18 @@ def test_constructor_success (self):
...
 
-    def test_distance_to_success_point (self):
-
-        line = Line(
-            Point(1.0, 2.0),
-            np.array((1.0, 1.0)),
-        )
+    def test_distance_to_success_point (
+        self,
+        line: Line
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_line.py:49]:
```suggestion
        line: Line,
```

File: bindings/python/test/geometry/d2/objects/test_line.py [hunk 16]
```diff
@@ -36,26 +36,18 @@ def test_constructor_success (self):
...
-        )
-
-        point_set = PointSet([
-            Point(1.0, 2.0),
-            Point(2.0, 3.0),
-        ])
+    def test_contains_success_point_set(
+        self,
+        line: Line,
+        point_set: PointSet
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_line.py:42]:
```suggestion
        point_set: PointSet,
```

File: bindings/python/test/geometry/d2/objects/test_linestring.py [hunk 17]
```diff
@@ -21,163 +21,164 @@
...
-# def test_geometry_d2_objects_linestring_to_string ():
+        assert isinstance(linestring_list, list)
+        assert isinstance(linestring_list[0], Point)
+        assert isinstance(linestring_list[1], Point)
+        assert len(linestring_list) == 2
 
-# def test_geometry_d2_objects_linestring_apply_transformation ():
+    # def test_geometry_d2_objects_linestring_to_string ():
+
+    # def test_geometry_d2_objects_linestring_apply_transformation ():
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_linestring.py:182]:
```suggestion
    # def test_to_string_success (self):

    # def test_apply_transformation_success (self):
```

File: bindings/python/test/geometry/d2/objects/test_multipolygon.py [hunk 18]
```diff
@@ -18,147 +18,108 @@
...
+        assert multipolygon.is_defined() is False
 
-    # Multiple MultiPolygons
-    multipolygon_1: MultiPolygon = MultiPolygon([square_1, square_2])
-    multipolygon_2: MultiPolygon = MultiPolygon([square_1])
-    multipolygon_3: MultiPolygon = MultiPolygon.polygon(square_1)
+    def test_getters_success (
+        self,
+        square_1: Polygon,
+        square_2: Polygon
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_multipolygon.py:102]:
```suggestion
        square_2: Polygon,
```

File: bindings/python/test/geometry/d2/objects/test_multipolygon.py [hunk 19]
```diff
@@ -18,147 +18,108 @@
...
-    assert multipolygon.get_polygon_count() == 1
-    assert multipolygon.get_polygons()[0] == square_1
-    assert multipolygon.get_convex_hull() == square_1
-
-# def test_geometry_d2_objects_multipolygon_union_with() :
-
-# def test_geometry_d2_objects_multipolygon_to_string ():
-
-# def test_geometry_d2_objects_multipolygon_apply_transformation ():
+    # def test_geometry_d2_objects_multipolygon_apply_transformation ():
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_multipolygon.py:123]:
```suggestion
    # def test_apply_transformation_success (self):
```

File: bindings/python/test/geometry/d2/objects/test_multipolygon.py [hunk 20]
```diff
@@ -18,147 +18,108 @@
...
-    assert multipolygon.is_defined() is False
+        assert multipolygon.get_polygon_count() == 1
+        assert multipolygon.get_polygons()[0] == square_1
+        assert multipolygon.get_convex_hull() == square_1
 
-# def test_geometry_d2_objects_multipoolygon_contains ():
+    # def test_geometry_d2_objects_multipolygon_union_with() :
 
-def test_geometry_d2_objects_multipoolygon_getters ():
+    # def test_geometry_d2_objects_multipolygon_to_string ():
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_multipolygon.py:121]:
```suggestion
    # def test_to_string_success (self):
```

File: bindings/python/test/geometry/d2/objects/test_multipolygon.py [hunk 21]
```diff
@@ -18,147 +18,108 @@
...
-    assert multipolygon is not None
-    assert isinstance(multipolygon, MultiPolygon)
-    assert isinstance(multipolygon, Object)
-    assert multipolygon.is_defined() is False
+        assert multipolygon.get_polygon_count() == 1
+        assert multipolygon.get_polygons()[0] == square_1
+        assert multipolygon.get_convex_hull() == square_1
 
-# def test_geometry_d2_objects_multipoolygon_contains ():
+    # def test_geometry_d2_objects_multipolygon_union_with() :
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_multipolygon.py:119]:
```suggestion
    # def test_union_with_success (self) :
```

File: bindings/python/test/geometry/d2/objects/test_point.py [hunk 22]
```diff
@@ -21,212 +21,213 @@
...
-    assert isinstance(point.to_string(), String)
-    assert point.to_string() == '[3.0, 5.0]'
+        point: Point = Point(3.0, 5.0)
 
-# def test_geometry_d2_objects_point_apply_transformation ():
+        assert isinstance(point.to_string(), String)
+        assert point.to_string() == '[3.0, 5.0]'
 
-#     point: Point = Point(4.5, 5.4)
+    # def test_geometry_d2_objects_point_apply_transformation ():
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_point.py:229]:
```suggestion
    # def test_apply_transformation_success (self):
```

File: bindings/python/test/geometry/d2/objects/test_polygon.py [hunk 23]
```diff
@@ -215,28 +215,19 @@ def test_comparators (self):
         assert polygon_2 == polygon_2
         assert polygon_1 != polygon_2
 
-    def test_intersects_polygon (self):
+    def test_intersects_polygon (
+        self,
+        square_1: Polygon,
+        square_2: Polygon,
+        square_3: Polygon,
+        square_4: Polygon
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_polygon.py:223]:
```suggestion
        square_4: Polygon,
```

File: bindings/python/test/geometry/d2/objects/test_polygon.py [hunk 24]
```diff
@@ -329,27 +320,62 @@ def test_getters (self):
 
         # get_convex_hull
 
-    def test_union_with (self):
+    def test_union_with (
+        self,
+        square_1: Polygon,
+        square_4: Polygon
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_polygon.py:326]:
```suggestion
        square_4: Polygon,
```

File: bindings/python/test/geometry/d2/objects/test_polygon.py [hunk 25]
```diff
@@ -267,8 +258,8 @@ def test_getters (self):
 
         # Inner Ring Vertices
         point_5: Point = Point(-0.5, 0.5)
-        point_6: Point = Point(0.5, 0.5)
-        point_7: Point = Point(0.5, -0.5)
+        point_6: Point = Point(0.5, -0.5)
+        point_7: Point = Point(0.5, 0.5)
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_polygon.py:262]: Why the change? (do we break the test otherwise?)

File: bindings/python/test/geometry/d2/objects/test_polygon.py [hunk 26]
```diff
@@ -329,27 +320,62 @@ def test_getters (self):
...
+        multipolygon: MultiPolygon = square_1.union_with(square_4)
 
         assert multipolygon is not None
         assert isinstance(multipolygon, MultiPolygon)
         assert multipolygon.is_defined()
 
+    def test_intersection_with (
+        self,
+        square_1: Polygon,
+        square_4: Polygon
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_polygon.py:338]:
```suggestion
        square_4: Polygon,
```

File: bindings/python/test/geometry/d2/objects/test_polygon.py [hunk 27]
```diff
@@ -329,27 +320,62 @@ def test_getters (self):
...
+        assert isinstance(intersection_2, Intersection)
+        assert intersection_2.is_defined()
+        assert intersection_2.is_line_string()
+        assert intersection_2.as_line_string() == LineString([Point(0.5, 0.5), Point(0.5, -0.5)])
+        assert intersection_2.get_type() == Intersection.Type.LineString
+
+    def test_difference_with (
+        self,
+        square_1: Polygon,
+        square_3: Polygon
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_polygon.py:362]:
```suggestion
        square_3: Polygon,
```

File: bindings/python/test/geometry/d2/objects/test_segment.py [hunk 28]
```diff
@@ -107,15 +96,24 @@ def test_getters_success (self):
 
         assert segment.get_length() == 2.0
 
-    def test_distance_to_success_point (self, segment: Segment):
+    def test_distance_to_success_point (
+        self,
+        segment: Segment
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_segment.py:101]:
```suggestion
        segment: Segment,
```

File: bindings/python/test/geometry/d2/objects/test_segment.py [hunk 29]
```diff
@@ -107,15 +96,24 @@ def test_getters_success (self):
...
+    ):
 
         assert segment.distance_to(Point(0.0, 0.0)) == 0.0
 
-    def test_distance_to_success_point_set (self, segment: Segment):
+    def test_distance_to_success_point_set (
+        self,
+        segment: Segment
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_segment.py:108]:
```suggestion
        segment: Segment,
```

File: bindings/python/test/geometry/d2/objects/test_segment.py [hunk 30]
```diff
@@ -107,15 +96,24 @@ def test_getters_success (self):
...
+    ):
 
         assert segment.distance_to(PointSet([Point(0.0, 0.0), Point(1.0, 0.0)])) == 0.0
 
-    def test_to_line_success (self, segment: Segment):
+    def test_to_line_success (
+        self,
+        segment: Segment
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_segment.py:115]:
```suggestion
        segment: Segment,
```

**Review by lucas-bremond — APPROVED** [review]

File: bindings/python/test/geometry/d2/objects/test_composite.py [hunk 31]
```diff
@@ -0,0 +1,118 @@
...
+        polygon: Polygon
+    ):
+
+        composite: Composite = Composite(polygon)
+
+        assert composite is not None
+        assert isinstance(composite, Composite)
+        assert composite.is_defined()
+
+    def test_geometry_d2_objects_composite_empty (self):
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_composite.py:36]:
```suggestion
    def test_empty_success (self):
```

File: bindings/python/test/geometry/d2/objects/test_composite.py [hunk 32]
```diff
@@ -0,0 +1,118 @@
...
+
+        composite: Composite = Composite.empty()
+
+        assert composite is not None
+        assert isinstance(composite, Composite)
+        assert isinstance(composite, Object)
+        assert composite.is_defined()
+        assert composite.is_empty()
+
+    def test_geometry_d2_objects_composite_defined (self):
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_composite.py:46]:
```suggestion
    def test_is_defined_success (self):
```

File: bindings/python/test/geometry/d2/objects/test_composite.py [hunk 33]
```diff
@@ -0,0 +1,118 @@
...
+        assert composite is not None
+        assert isinstance(composite, Composite)
+        assert isinstance(composite, Object)
+        assert composite.is_defined() is False
+
+    def test_geometry_d2_objects_composite_comparators (
+        self,
+        composite: Composite,
+        polygon: Polygon
+    ):
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_composite.py:59]:
```suggestion
    def test_comparators_success (
        self,
        composite: Composite,
        polygon: Polygon,
    ):
```

File: bindings/python/test/geometry/d2/objects/test_composite.py [hunk 34]
```diff
@@ -0,0 +1,118 @@
...
+
+        assert composite == composite
+        assert composite != other_composite
+        assert other_composite != composite
+
+    def tes_geometry_d2_objects_composite_addition_operator (
+        self,
+        composite: Composite,
+        polygon: Polygon
+    ):
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_composite.py:71]:
```suggestion
    def test_addition_operator_success (
        self,
        composite: Composite,
        polygon: Polygon,
    ):
```

File: bindings/python/test/geometry/d2/objects/test_composite.py [hunk 35]
```diff
@@ -0,0 +1,118 @@
...
+        assert composite.get_object_count() == 4
+
+        with pytest.raises(RuntimeError):
+            composite += Composite.undefined()
+
+    def test_geometry_d2_objects_composite_is (
+        self,
+        composite: Composite,
+        polygon: Polygon
+    ):
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_composite.py:97]:
```suggestion
    def test_is_success (
        self,
        composite: Composite,
        polygon: Polygon,
    ):
```

File: bindings/python/test/geometry/d2/objects/test_composite.py [hunk 36]
```diff
@@ -0,0 +1,118 @@
...
+        assert composite.is_polygon() is False
+
+        assert Composite(polygon).is_polygon()
+
+    def test_geometry_d2_objects_composite_access_object_at (
+        self,
+        composite: Composite,
+        polygon: Polygon,
+        point: Point
+    ):
```
Comment (lucas-bremond) [inline bindings/python/test/geometry/d2/objects/test_composite.py:109]:
```suggestion
    def test_access_object_at_success (
        self,
        composite: Composite,
        polygon: Polygon,
        point: Point,
    ):
```

**Review by lucas-bremond — APPROVED** [review]

File: bindings/python/test/geometry/d2/objects/test_polygon.py (hunk 25 above)
Comment (Derollez) [reply bindings/python/test/geometry/d2/objects/test_polygon.py:262]: Yes we do, this is linked to the ring ordering which is expected in boost to have valid polygons and that we don't enforce. We can find a way to keep the flexibility on our side while converting when needed and going through boost. I have a ticket to check/tackle that one separately
